Refactor/reviewsAndCleansAllCode

cordhernandez opened the pull request:
cleans all code 

This PR resolves two tickets
#126 
#122

cordhernandez committed:
organizes Arrays+.swift

adds spaces between lines

cordhernandez committed:
cleans code in DistanceCalculation.swift

renames variables

cordhernandez committed:
adds goCombineAndLoadAddress()

adds method that combines and loads store addresses

cordhernandez committed:
adds Loading Indicator extension

cordhernandez committed:
removes struct AddressInfo

This struct is never used
Also removes constant location (never used)

cordhernandez committed:
fixes grammar

cordhernandez committed:
organizes code

adds spaces between lines

cordhernandez committed:
cleans/organizes code

adds spaces in between lines

SirWellington commented on BlackNectar/BlackNectar/Arrays+.swift:
```diff
@@ -14,12 +14,13 @@ extension Array {
-        Tells whether the specified index is in bounds of the array.
-        Use this check with indexes before accessing the array.
-    */
+// Tells whether the specified index is in bounds of the array.
```
It is better to leave it with the ``/**`. Why, because of code documentation:

http://nshipster.com/swift-documentation/

cordhernandez replied:
fixed

SirWellington commented on BlackNectar/BlackNectar/Arrays+.swift:
```diff
@@ -14,12 +14,13 @@ extension Array {
-    */
+// Tells whether the specified index is in bounds of the array.
+// Use this check with indexes before accessing the array.
+
```
Same here.

cordhernandez replied:
fixed

SirWellington commented on BlackNectar/BlackNectar/DistanceCalculation.swift:
```diff
@@ -14,13 +14,13 @@ class DistanceCalculation {
+        let storeCoordinate = CLLocation(latitude: storeLocation.latitude, longitude: storeLocation.longitude)
         
-        let calculatedMeters = userCoordinateToLocation.distance(from: storeCoordinateToLocation)
+        let calculateMeters = userCoordinate.distance(from: storeCoordinate)
```
`distanceInMeters`

cordhernandez replied:
fixed

SirWellington commented on BlackNectar/BlackNectar/StoresDataStruct.swift:
```diff
@@ -25,7 +25,6 @@ struct StoresInfo {
     static func fromJson(dictionary: NSDictionary) -> StoresInfo? {
         
         guard let storeName = dictionary ["store_name"] as? String,
-            let location = dictionary ["location"] as? NSDictionary,
```
Are we not using it? Wondering why we removed it.

cordhernandez replied:
it looks like let coordinatesDictionary is being in replace

cordhernandez replied:
guard let coordinatesDictionary = dictionary ["location"] as? NSDictionary else { return nil }

SirWellington commented on BlackNectar/BlackNectar/StoresMapViewController.swift:
```diff
@@ -99,7 +102,7 @@ extension StoresMapViewController {
     func loadStoresInMapView(at coordinate: CLLocationCoordinate2D) {
         
-        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        networkLoadingIndicatorIsSpinning()
```
Again, this function should just be renamed to `spinNetworkIndicator` or `showSpinner()`

cordhernandez replied:
done

SirWellington commented on BlackNectar/BlackNectar/StoresMapViewController.swift:
```diff
@@ -295,5 +298,23 @@ extension StoresMapViewController {
+        
+    }
+    
+    func networkLoadingIndicatorIsNotSpinning() {
```
`stopSpinningIndicator()` or `stopNetworkSpinner()`

cordhernandez replied:
done

SirWellington commented on BlackNectar/BlackNectar/StoresTableViewController.swift:
```diff
@@ -74,6 +70,23 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate,
         }
     }
     
+    private func filterStoresFrom(stores: [StoresInfo]) -> [StoresInfo] {
```
I know I wrote this, but it is probably better named:
`filterStores(from stores: [StoreInfo]) -> [StoresInfo]`

cordhernandez replied:
done

SirWellington commented on BlackNectar/BlackNectar/StoresTableViewController.swift:
```diff
@@ -129,6 +126,17 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate,
+    
+    func goCombineAndLoadAddress(into cell: StoresTableViewCell, withStore address: NSDictionary) {
+        
+        guard let street = address["address_line_1"] as? String else { return }
```
To be honest, the Address should have its own Data Structure, and a Store should have an Address type.
That is better than constantly having to pull out the values from the dictionary each time.

Hit me up if you need clarification on this.

SirWellington reviewed (commented):
That code tho! 😍  Lit!
Few comments here and there, but overall love the spirit of this.

cordhernandez committed:
adds comments

cordhernandez committed:
cleans code

renames calculateMeters to distanceInMeters

cordhernandez committed:
cleans code

renames networkLoadingIndicatorIsSpinning() to startSpinningIndicator()
renames networkLoadingIndicatorIsNotSpinning() to
stopSpinningIndicator()

cordhernandez committed:
cleans code

renames method filterStoresFrom() filterStores()

cordhernandez committed:
cleans code

changes StoresInfo to Stores
changes StoresInfo.fromJson to Stores.getStoreJsonData

cordhernandez committed:
adds Address Struct and cleans code

adds Address Struct
changes fromJson() to getStoreJsonData()
changes struct StoresInfo to struct Stores
modifies the LOG.error description

cordhernandez committed:
changes StoresInfo to Stores

cordhernandez committed:
Modifies goCombineAndLoadAddress

cleans code
modifies parameters

SirWellington commented on BlackNectar/BlackNectar/StoresDataStruct.swift:
```diff
@@ -11,63 +11,65 @@ import CoreLocation
-// Creating data structure for JSON Request
-struct StoresInfo {
+// Creates data structure for JSON Request
+struct Stores {
```
Nice! Well done. Except it should be `Store`, in the singular.
Remember, we declare `Int`, `Double`, etc, and not `Ints`, `Doubles`. That would be an array.

cordhernandez replied:
done and done 
thanks

SirWellington commented on BlackNectar/BlackNectar/StoresDataStruct.swift:
```diff
@@ -11,63 +11,65 @@ import CoreLocation
         }
         
-        guard let coordinatesDictionary = dictionary ["location"] as? NSDictionary else { return nil }
+        guard let addressObject = Address(from: addressJSON) else { return nil }
```
nice! Though you could probably just call this variable `address`

cordhernandez replied:
got it! 
done!

SirWellington commented on BlackNectar/BlackNectar/StoresDataStruct.swift:
```diff
@@ -11,63 +11,65 @@ import CoreLocation
 }
 
-struct AddressInfo {
+struct Address {
```
Very good!

cordhernandez replied:
thank you sir

SirWellington commented on BlackNectar/BlackNectar/StoresDataStruct.swift:
```diff
@@ -11,63 +11,65 @@ import CoreLocation
-    let street: String
-    let city: String
-    let state: String
+    let addressLineOne: String?
```
hmmm these fields should probably not be optionals. The only one of these that is really optional in an address is the `addressLineTwo`.

cordhernandez replied:
Having issues with this one 
At the moment the only way I can make it work is if I make all of them optionals
could use some help

SirWellington commented on BlackNectar/BlackNectar/StoresTableViewController.swift:
```diff
@@ -129,6 +126,17 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate,
         
     }
+    
+    func goCombineAndLoadAddress(into cell: StoresTableViewCell, withStore store: Stores) {
```
This can now be renamed to just `insertAddress(...)`

cordhernandez replied:
👌🏾

SirWellington reviewed (commented):
Wayyy betta

cordhernandez committed:
Changes Stores to Store

cordhernandez committed:
changes Stores to Store

changes goCombineAndLoadAddress() into insertAddress()

cordhernandez committed:
changes Stores to Store

cordhernandez committed:
cleans code

changes AddressObject to address

Commit:
Renames folder Model.

That’s what it represents.

Commit:
Removes optionals from Address object

Commit:
Adds initializer to Store that accepts JSON

Commit:
Cleans code

Commit:
Cleans code

Commit:
Moves aroma message

cordhernandez commented:
Everything looks good! Thanks for your help. Merging!
